src/web/handlers.rs

```diff
@@ -187,12 +170,8 @@ where
                 .map_err(From::from)
                 .map(move |ts| Ok((result, ts)))
         })
-        .map(
-            move |r: std::result::Result<
-                (Paginated<T>, std::result::Result<SyncTimestamp, Error>),
-                Error,
-            >| {
-                let (result, ts) = r.expect("Could not get r in finish_get_collection");
+        .map_ok(
+            move |(result, ts): (Paginated<T>, std::result::Result<SyncTimestamp, Error>)| {
                 let ts = ts.expect("Could not get ts in finish_get_collection");
```

Comment:
Forgot this last `expect()`

Reply:
No, I removed the expect for r but ts is still a result (both r and ts were results previously).